src/types.py

```diff
@@ -1,39 +1,99 @@
 import copy
 from dataclasses import dataclass, field
-from typing import List, Optional, Set, Tuple, Union
+from typing import Dict, List, Optional, Set, Tuple, Union
 
 import pycparser.c_ast as ca
 
 from .c_types import (
     CType,
     Struct,
     TypeMap,
-    equal_types,
-    get_struct,
     parse_constant_int,
     parse_function,
     parse_struct,
     primitive_size,
     resolve_typedefs,
     set_decl_name,
     to_c,
-    var_size_align,
 )
 from .error import DecompFailure
 from .options import Formatter
 
+AccessPath = List[Union[str, int]]
+
+
+@dataclass(eq=False)
+class TypeUniverse:
```

Comment:
This is intended to be a *mutable* set of `Type`s and contains information about derived types. This also completely abstracts `TypeMap` away; `translate.py` no longer references `typemap` at all. 

It keeps a reference to the `TypeMap`, which solely represents the information in the context file.
For example, when running tests, each test runs with a fresh `TypeUniverse` object, but they share `TypeMap`s. 

I don't love the name though :smile: -- not sure if anyone has any better ideas. One option is to rename `c_types.TypeMap` to something like `ContextTypeMap`/`CTypeMap`/`Context`, so this class could take the `TypeMap` name.